R/ParamSetBase.R

```diff
@@ -86,6 +86,17 @@ ParamSetBase = R6Class("ParamSetBase",
     }
   ),
   private = list(
-    priv_dictionary = NULL
+    priv_dictionary = NULL,
+    deep_clone = function(name, value) {
+      if(name == "params") {
+        lapply(value, function(x) {
+          x$clone()
```

Comment:
I don't know why I can't put here `deep = TRUE`, because it causes the infinite recursion.